platform/security/platform-security-core-api/src/main/java/ddf/security/SubjectUtils.java

```diff
@@ -307,6 +308,21 @@ public static String getEmailAddress(Subject subject) {
                 }));
   }
 
+  /**
+   * Get a subject's unique identifier. If email is present, then that is the unique identifier. If
+   * not, user name is returned.
+   *
+   * @param subject
+   * @return identifier or null if not found.
+   */
+  public static String getUniqueIdentifier(Subject subject) {
+    String identifier = getEmailAddress(subject);
```

Comment:
❓ I don't have any experience in this code, but is it possible for two `Subject`s to have the same email address?

Reply:
Email is supposed to be unique.